### PR TITLE
Text-element render sub-range

### LINKED ISSUE #1584: Add ability to render a sub-range of text to the text-element
With this it will be possible to efficiently add "typie" text effects.

### PR DESCRIPTION
**Pull request by codebon77**
Fixes #1584

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/components/element/text-element.js [hunk 1]
```diff
@@ -1120,6 +1107,53 @@ Object.assign(pc, function () {
         _shouldAutoFit: function () {
             return this._autoFitWidth && !this._autoWidth ||
                    this._autoFitHeight && !this._autoHeight;
+        },
+
+        // calculate the number of characters per texture up to, but not including
+        // the specified symbolIndex
+        _calculateCharsPerTexture: function (symbolIndex) {
```
Comment (daredevildave) [inline src/framework/components/element/text-element.js]: symbolIndex arg doesn't seem to be used can we remove it?

Comment (codebon77) [reply src/framework/components/element/text-element.js]: Not sure I understand... symbolIndex index is used in the function and two of the callsites...

**Comment by willeastcott** [comment]
Can you add some API ref docs for the two new properties please? 😄

**Comment by codebon77** [comment]
text-element.js currently has no API docs at all. Should I add the API docs as private or..?

**Comment by willeastcott** [comment]
Oh, looks like you haven't exposed the properties to pc.ElementComponent. Look at, say, enableMarkup for an example.

**Comment by willeastcott** [comment]
And the docs for these properties should be public.

**Pull request merged**